.codeclimate.yml

```diff
@@ -0,0 +1,43 @@
+version: "2"         # required to adjust maintainability checks
+checks:
+  # Methods or functions defined with a high number of arguments
+  argument-count:
+    config:
+      threshold: 6
+  # Boolean logic that may be hard to understand
+  complex-logic:
+    config:
+      threshold: 4
+  # Excessive lines of code within a single file
+  file-lines:
+    config:
+      threshold: 750
+  # Functions or methods that may be hard to understand. This is disabled as it doesn't appear to be a reliable metric.
+  method-complexity:
+    enabled: false
+  # Classes defined with a high number of functions or methods. 
+  method-count:
+    config:
+      threshold: 20
+  # Excessive lines of code within a single function or method
+  method-lines:
+    config:
+      threshold: 500
+  #  Deeply nested control structures like if or case
+  nested-control-flow:
+    config:
+      threshold: 4
+  # Functions or methods with a high number of return statements. This is disabled because Go encourages early returns.
+  return-statements:
+    enabled: false
+  # Duplicate code which is not identical but shares the same structure (e.g. variable names may differ). This is disabled due to too many false trips.
+  similar-code:
+    enabled: false
+  # Code that is identical in structure
+  identical-code:
+    config:
+      threshold: # language-specific defaults. an override will affect all languages.
+exclude_paths:
+  - vendor/
+  - static/
+  - t/
```

Comment:
Is it possible to exclude individual paths? It would be nice to have `t/` covered, unless it's too many errors.

Reply:
Let me try to ignore some of the files that were causing some false trips with the identical code metric

Reply:
@adambabik my idea was to just ignore identical code in the `t/` directory, but I can't seem to get the configuration for that to work. You can take a look at the errors CodeClimate is finding in t/, I personally don't think identical code like that is a big deal in tests

Reply:
Definitely not an issue in tests. Unfortunately, I guess it's not possible to disable a single check for specific files, righ? Feel free to make it just pass and we can merge it.

Reply:
@adambabik Unforunately, I couldn't get the per-check pattern ignoring configuration to work :( I'll revert it back. It's a good starting point at least :+1: